core/commands/setup_dependencies_command.rb

```diff
@@ -315,13 +356,32 @@ def install_dependencies
                             "sudo DEBIAN_FRONTEND=noninteractive apt-get -yq install #{required_packages.join(' ')}",
                             'sudo systemctl restart libvirtd.service'
                           ])
-    return result[:value].exitstatus unless result[:value].success?
+    product = @env.nodeProduct
+    if product == 'docker' || product.nil?
```

Comment:
Please add a method called `should_install?(product)` that will answer true whether the installation is needed or not.